### ISSUE TITLE
Pre-Determine tunnels for apps

### ISSUE THREAD

**Issue by bijeebuss** [Radiokot/umbrel-cloudflared#6]
I had an idea for how to potentially use this to solve an issue with [trying to setup fedimint as an umbrel app.](https://github.com/fedimint/fedimint/issues/4242) but I think it could also be relevant for other apps. Fedimint needs to be available publicly and it needs to know what URL its available on before even starting up. 

What are your thoughts on this change?: 

This app could have a UI section where you associate one of your tunnels with an umbrel app ID. Then it writes the URL to a file somewhere in the app data like `/data/umbrel-apps/my_umbrel_app_id` and the contents of that file would just be the public URL you will be pointing to the app. This way when the fedimint app starts up it can mount the cloudflared app data dir somewhere and then read its URL from the file based on its app ID.

Or maybe there is an easier way to achieve this by setting a metadata flag on the tunnel or something and then using curl to get it?

**Comment by Radiokot** [comment]
This app is a tunnel connector, it can't manage the routes. The routes are managed under your Cloudflare Zero Trust dashboard. If I correctly unredstood what you are asking for, I think it can't be done. 

If I had a service requiring some preparation before starting up, I would:
- Create a configuration service with a UI to make all the preparations;
- Add this configuration service to the compose with custom `healthcheck` for it, so it reports healthy status only if successfully configured;
- Make the main service dependent on the configuration with `depends_on`.

**Comment by bijeebuss** [comment]
It can't manage the routes but it can list them right? That would be sufficient to do this. But either way I think like your idea better anyway

**Comment by Radiokot** [comment]
Good luck 😉

**Comment by bijeebuss** [comment]
Do you know any good examples of apps that do this already?

**Comment by Radiokot** [comment]
No, just had this idea. This is a creative hacking/patching when you can't modify the original software 🙂
